lib/datadog/tracing/contrib/utils/quantization/http.rb

```diff
@@ -14,20 +14,25 @@ module HTTP
 
             PLACEHOLDER = '?'.freeze
 
+            # taken from Ruby 3.1.2 lib/uri/rfc3986_parser.rb and adjusted to stop at path
+            RFC3986_URL_BASE = /\A(?<URI>(?<scheme>[A-Za-z][+\-.0-9A-Za-z]*):(?<hier-part>\/\/(?<authority>(?:(?<userinfo>(?:%\h\h|[!$&-.0-;=A-Z_a-z~])*)@)?(?<host>(?<IP-literal>\[(?:(?<IPv6address>(?:\h{1,4}:){6}(?<ls32>\h{1,4}:\h{1,4}|(?<IPv4address>(?<dec-octet>[1-9]\d|1\d{2}|2[0-4]\d|25[0-5]|\d)\.\g<dec-octet>\.\g<dec-octet>\.\g<dec-octet>))|::(?:\h{1,4}:){5}\g<ls32>|\h{1,4}?::(?:\h{1,4}:){4}\g<ls32>|(?:(?:\h{1,4}:)?\h{1,4})?::(?:\h{1,4}:){3}\g<ls32>|(?:(?:\h{1,4}:){,2}\h{1,4})?::(?:\h{1,4}:){2}\g<ls32>|(?:(?:\h{1,4}:){,3}\h{1,4})?::\h{1,4}:\g<ls32>|(?:(?:\h{1,4}:){,4}\h{1,4})?::\g<ls32>|(?:(?:\h{1,4}:){,5}\h{1,4})?::\h{1,4}|(?:(?:\h{1,4}:){,6}\h{1,4})?::)|(?<IPvFuture>v\h+\.[!$&-.0-;=A-Z_a-z~]+))\])|\g<IPv4address>|(?<reg-name>(?:%\h\h|[!$&-.0-9;=A-Z_a-z~])*))(?::(?<port>\d*))?)))(?:\/|\z)/.freeze # rubocop:disable Style/RegexpLiteral, Layout/LineLength
```

Comment:
I suggest:

1. pointing at the latest canonical version (since this gem has been extracted into a separate repo that then gets syncd into Ruby) https://github.com/ruby/uri/blob/master/lib/uri/rfc3986_parser.rb
2. Including the copyright header for the library here, as well as mentioning this in `LICENSE-3rdparty.csv`
3. Adding a comment being more specific about what "adjusted to stop at path" means (e.g. to make maintenance easier in the future)

Reply:
> since this gem has been extracted into a separate repo that then gets syncd into Ruby

Good point.

> pointing at the latest canonical version (since this gem has been extracted into a separate repo that then gets syncd into Ruby)

Then I would rather point at the commit from which it was taken to make it a permalink.